### PR TITLE
Pr from fork/1034

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR is for @homerdin 's RAJA SYCL back-end development. The original PR from his forked repo is here: https://github.com/LLNL/RAJA/pull/1034

Please review and approve this PR for merging.

### CODE REVIEW

**Comment by rhornung67** [comment]
@homerdin here is your PR branch pulled into the RAJA repo. I made a couple changes to disable SYCL by default (enable by passing '-DENABLE_SYCL=On' to CMake and add missing include guards to prevent SYCL headers from being pulled in if SYCL is not enabled. Hopefully, this will make all our non-SYCL CI checks pass. We're working on building a container image of the open-source intel compiler to build with SYCL in our Azure CI.

**Comment by rhornung67** [comment]
Folks, it looks like my changes are on track to get our non-SYCL CI checks to pass (fingers crossed). Now, we have to find a way to build and run SYCL CI tests....

**Comment by mdavis36** [comment]
@rhornung67 I haven't had a go at it yet but I can take a look into getting the axoms sycl docker running this branch on our azure today

**Comment by rchen20** [comment]
@mdavis36 If you run in to any linking warnings, the bugfix/chen59/sycllinking branch has some hacks for finding shared libraries.

File: CMakeLists.txt
Comment (davidbeckingsale) [inline CMakeLists.txt:238]: Where does this sycl target come from? Or is this just used so that CMake will generate `-lsycl`?

**Comment by rhornung67** [comment]
@homerdin all non-SYCL checks have come back green after some minor adjustments. So all we need is approvals on this before merging.  We are exploring our options to run CI checks with the SYCL back-end. We don't have a supported platform for this at Livermore. Do you have any experience with the open-source llvm-based intel compiler? We could set up a container image and run CI on azure with that.

File: CMakeLists.txt
Comment (homerdin) [reply CMakeLists.txt:238]: It looks like this is unnecessary currently.  All that is needed to build is the `-fsycl` and `-fsycl-unnamed-lambda` compiler flags.  It wasn't clear if there would be support in cmake or blt when it was added.

**Comment by homerdin** [comment]
@rhornung67 Sounds good.  I've built/installed the open source llvm-based intel compiler.  I'd be happy to help get a setup running.

**Comment by rhornung67** [comment]
@mdavis36 if you have any questions about getting the open-source intel compiler image set up and working in our azure configuration, please feel free to reach out to @homerdin (see his comment above).

File: examples/tut_add-vectors.cpp
Comment (artv3) [inline examples/tut_add-vectors.cpp:59]: What does this do?

File: examples/tut_dot-product.cpp
Comment (artv3) [inline examples/tut_dot-product.cpp:197]: When is `sycl_exec_nontrivial` needed? is it just reductions?

File: examples/tut_add-vectors.cpp
Comment (rhornung67) [reply examples/tut_add-vectors.cpp:59]: The camp resources do memory alloc/dalloc/copy, etc. We use those in the parameterized RAJA tests.

File: examples/tut_dot-product.cpp
Comment (rhornung67) [reply examples/tut_dot-product.cpp:197]: They are needed for any kernel that contains anything that is not-trivially copy-able. @homerdin added a note to the policy section of the user guide about this. We are encouraging users to start with the "nontrivial" versions of the policies to make sure their code works. I believe @homerdin was exploring some C++ std machinery to see if this can be made simpler for users.

**Review by artv3 — DISMISSED** [review]

**Comment by rhornung67** [comment]
Folks, please check in on this PR occasionally and reapprove if you approved earlier. Changes have been made that have dismissed previous approvals.  Thank you.

**Review by mdavis36 — APPROVED** [review]

**Review by artv3 — DISMISSED** [review]

**Review by rchen20 — APPROVED** [review]

**Comment by rhornung67** [comment]
@rchen20 thanks for the camp update. travis is still not reporting. we will discuss at this morning's meeting.

File: src/MemUtils_SYCL.cpp
Comment (MrBurmark) [inline src/MemUtils_SYCL.cpp:49]: It looks like this still needs to be fixed, unless we don't allow openmp and sycl together.

File: include/RAJA/policy/sycl/MemUtils_SYCL.hpp
Comment (MrBurmark) [inline include/RAJA/policy/sycl/MemUtils_SYCL.hpp:63]: This declaration is incompatible with the definition when openmp is also enabled.

File: src/MemUtils_SYCL.cpp
Comment (rhornung67) [reply src/MemUtils_SYCL.cpp:49]: @MrBurmark Let's go with this for now.  I'll add an issue to address later.

File: include/RAJA/policy/sycl/MemUtils_SYCL.hpp
Comment (rhornung67) [reply include/RAJA/policy/sycl/MemUtils_SYCL.hpp:63]: WIll also include this in the issue.

**Pull request merged**
